babel/board.js

```diff
@@ -13,19 +13,22 @@ let SortedSet = require('collections/sorted-set');
  *   [ [6, 3, 6], [5, 2, 4] ], [ [5, 2, 4], [4, 2, 0] ] ]
  */
 export function iterchunks (orbs, chunkLimitRange = [4, 2]) {
-    // Defaults will find all possible 3-matches. Used by `Board#evaluate`.
-    let [width, height] = chunkLimitRange;
-    let [finalPositionWidth, finalPositionHeight] = [orbs[0].length - width, orbs.length - height];
-    let chunks = [];
-    _.each(_.range(0, finalPositionHeight + 1), heightIndex => {
-        _.each(_.range(0, finalPositionWidth + 1), widthIndex => {
-            chunks.push([
-                orbs[heightIndex].slice(widthIndex, widthIndex + width),
-                orbs[heightIndex + 1].slice(widthIndex, widthIndex + width)
-            ]);
+    let _iterchunks = (orbs, chunkLimitRange) => {
+        let chunks = [];
+        let [width, height] = chunkLimitRange;
+        let [finalPositionWidth, finalPositionHeight] = [orbs[0].length - width, orbs.length - height];
+        _.each(_.range(0, finalPositionHeight + 1), heightIndex => {
+            _.each(_.range(0, finalPositionWidth + 1), widthIndex => {
+                chunks.push([
+                    orbs[heightIndex].slice(widthIndex, widthIndex + width),
+                    orbs[heightIndex + 1].slice(widthIndex, widthIndex + width)
+                ]);
+            });
         });
-    });
-    return chunks;
+        return chunks;
+    };
+    let transposedOrbs = _.zip(...orbs);
+    return [..._iterchunks(orbs, chunkLimitRange), ..._iterchunks(transposedOrbs, chunkLimitRange)]
```

Comment:
Okay, this is pretty interesting. You return `chunks` in line 28, which ends the `_iterchunks` function that you tucked inside `iterchunks`. THEN, you return and array of two calls of `_iterchunks`.